manim/mobject/svg/tex_mobject.py

```diff
@@ -113,6 +113,37 @@ def construct(self):
             tex.set_color_by_tex('igsta', RED)
             self.add(tex)
 
+Note that :func:`~.set_color_by_tex`: colors the entire substring containing the Tex searched for,
+not just the specific symbol or Tex expression searched for. Consider the following example:
+
+.. manim:: IncorrectLaTeXSubstringColoring
+    :save_last_frame:
+
+    class IncorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                "e^x = x^0 + x^1 + \\frac{1}{2} x^2 + \\frac{1}{6} x^3 + \\cdots + \\frac{1}{n!} x^n + \\cdots", isolate="x"
+                )
+            xs = equation.get_parts_by_tex("x")
+            xs.set_color(YELLOW)
+            self.add(equation)
+
+colors the entire equation yellow, contrary to what may be expected. To color only `x` yellow, we have to do the following:
```

Comment:
```suggestion
colors the entire equation yellow, contrary to what may be expected. To color only ``x`` yellow, we have to do the following:
```

Reply:
Also, this doesn't read like a proper sentence together with the part before the example:

"Consider the following example: ... colors the entire equation yellow ..."

I'd rather go for something like

```
Consider the following example.

...

As you can see, with this approach the entire equation is colored yellow, contrary to what may me expected. ...
```